### ISSUE TITLE
The latest tippecanoe 2.51.0 cannot produce vector tiles well

### ISSUE THREAD

**Issue by JinIgarashi** [felt/tippecanoe#219]
I am not sure this is a bug, but the latest tippecanoe does not produce tiles well.

In my local tippecanoe `2.24.0`, the following command can work well.

The below is PMTiles produced by 2.24.0

[nanyuki.pmtiles - tippecanoe 2.24.0.zip](https://github.com/felt/tippecanoe/files/14627939/nanyuki.pmtiles.-.tippecanoe.2.24.0.zip)

![image](https://github.com/felt/tippecanoe/assets/2639701/77888876-3fab-47d8-8bd9-397b34c94d7a)

<details>
<summary>v2.24.0 logs</summary>

```
tippecanoe     --no-feature-limit     --simplify-only-low-zooms     --detect-shared-borders     --read-parallel     --no-tile-size-limit     --no-tile-compression     --force     --name="NAWASCO Vector Tiles"      --description="Vector tiles for water & sanitation data in Nanyuki Water and Sanitation company, Kenya"     --attribution="©Nanyuki Water and Sanitation Co., Ltd."     -o /tmp/src/data/nanyuki.mbtiles /tmp/src/pipeline.geojson /tmp/src/sewerline.geojson /tmp/src/meter.geojson /tmp/src/kiosks.geojson /tmp/src/borehole.geojson /tmp/src/tank.geojson /tmp/src/booster_pump.geojson /tmp/src/chambers.geojson /tmp/src/manholes.geojson /tmp/src/t_wrks.geojson /tmp/src/blocks.geojson /tmp/src/blocks_annotation.geojson /tmp/src/cadastral.geojson /tmp/src/cadastral_annotation.geojson

60043 features, 3120496 bytes of geometry, 666045 bytes of string pool
  99.9%  14/9882/8193  /0  
```

</details>

But, in the latest 2.51.0, it shows a message like `60043 features, 3355810 bytes of geometry and attributes, 666045 bytes of string pool, 0 bytes of vertices, 0 bytes of nodes` while there is no `0 bytes of vertices, 0 bytes of nodes` log in v2.24.0.

The below is PMTiles produced by 2.51.0. As you can see, most of features such as pipeline, sewerline are not in tiles.

I think my tippecanoe command might have problems, but I am struggling to find a solution to fix this. I am not sure this problem is there since which version. Do you have any idea to fix the problem?

[nanyuki.pmtiles - tippecanoe 2.51.0.zip](https://github.com/felt/tippecanoe/files/14627944/nanyuki.pmtiles.-.tippecanoe.2.51.0.zip)

![image](https://github.com/felt/tippecanoe/assets/2639701/f1ea22fa-d26d-48a2-9f25-f7a35e199199)

<details>
<summary>v2.51.0 logs</summary>

```
tippecanoe     --no-feature-limit     --simplify-only-low-zooms     --detect-shared-borders     --read-parallel     --no-tile-size-limit     --no-tile-compression     --force     --name="NAWASCO Vector Tiles"      --description="Vector tiles for water & sanitation data in Nanyuki Water and Sanitation company, Kenya"     --attribution="©Nanyuki Water and Sanitation Co., Ltd."     -o /tmp/src/data/nanyuki.pmtiles /tmp/src/pipeline.geojson /tmp/src/sewerline.geojson /tmp/src/meter.geojson /tmp/src/kiosks.geojson /tmp/src/borehole.geojson /tmp/src/tank.geojson /tmp/src/booster_pump.geojson /tmp/src/chambers.geojson /tmp/src/manholes.geojson /tmp/src/t_wrks.geojson /tmp/src/blocks.geojson /tmp/src/blocks_annotation.geojson /tmp/src/cadastral.geojson /tmp/src/cadastral_annotation.geojson

60043 features, 3355810 bytes of geometry and attributes, 666045 bytes of string pool, 0 bytes of vertices, 0 bytes of nodes
  99.9%  14/9877/8190  /0  
```
</details>

I attach geojson files here to reproduce.

[geojson.zip](https://github.com/felt/tippecanoe/files/14627972/geojson.zip)

**Comment by e-n-f** [comment]
Thanks again. The bug appears to have been introduced in version 2.43.0. I'll take a look at that PR and figure out what I did wrong.

**Comment by e-n-f** [comment]
This will be fixed in https://github.com/felt/tippecanoe/pull/221

**Comment by e-n-f** [comment]
Thanks! Merged and tagged as 2.52.0.
